codegens/csharp-restsharp/lib/restsharp.js

```diff
@@ -30,6 +30,19 @@ function makeSnippet (request, options) {
   }
   snippet += `var request = new RestRequest(${isUnSupportedMethod ? '' : ('Method.' + request.method)});\n`;
   snippet += parseRequest.parseHeader(request.toJSON(), options.trimRequestBody);
+  if (request.body && request.body.mode === 'formdata') {
+    let isFile = false;
+    request.body.toJSON().formdata.forEach((data) => {
+      if (!data.disabled && data.type === 'file') {
+        isFile = true;
+      }
+    });
+    // The following statement needs to be added else the multipart/form-data request where there is no file
+    // is being sent as x-www-form-urlencoded by default
+    if (!isFile) {
```

Comment:
Is this necessary to do? 
Will it be okay if we use  AlwaysMultipartFormData for all kind of formdata requests?

Reply:
whenever there is a file, this parameter isn't necessary. It converts to multipart/form-data itself. This parameter only needs to be added when you have only text fields.